content/blog/elevate-your-v-project-with-unit-tests.md

```diff
@@ -0,0 +1,749 @@
+   while being confident that things will be fine as long as their code passes
+   all the unit tests.
+4. Running a unit test (ideally) does not take a long time. This **speeds up**
+   **the iteration process** and helps developers ship their software faster.
+5. Unit tests act as a supplement to the documentation and illustrative
+   examples. They outline the behaviour of the system, and provide
+   **concrete examples** of how to use the corresponding APIs.
+
+You can take a look at the Wikipedia article on
+[Unit testing](https://en.wikipedia.org/wiki/Unit_testing)
+to learn more about it.
+
+## V's Testing Culture
+
+I joined the V team in 2021. Before that, I had contributed to
+the [`rand`](https://modules.vlang.io/rand.html) module along with other devs
+which involved writing a lot of unit tests.
+
+It was rather simple to do because all the developers before us had included
+a lot of unit tests in vlib that we could use as a reference. This is still valid
+today. Whenever we propose the addition of new code to V's standard library,
+we make sure to include unit tests for it. Additionally, before merging code
+into the main repository, the GitHub Actions CI runner runs a whole bunch of
+tests on it, including all the unit tests. This helps make sure that new code
+does not break any existing code.
+
+This disciplined manner of software development is good to emulate in all V
+projects.
+
+## Prerequisites
+
+We assume that you know your way around the command-line interface. If
+you don't, here a good article to help you get started:
+[The Command Line for Complete Beginners](https://flaviocopes.com/cli-for-beginners/)
+
+Prior exposure to V is required. You can follow this article
+for a quickstart on V: [Getting Started with V](/getting-started-with-v/).
+
+We will pick up from where we left off in the
+[previous article](/the-complete-beginners-guide-to-cli-apps-in-v/),
+so make sure to have your project set up and ready to go.
+
+Alternatively, you can clone the
+[`geo` repository](https://github.com/hungrybluedev/geo), delete the
+`test` directory and follow along with the tutorial. More information
+on V's support for testing can be found in the
+[documentation](https://github.com/vlang/v/blob/master/doc/docs.md#testing).
+
+## A Brief Recap of `geo`
+
+In the [last tutorial](/the-complete-beginners-guide-to-cli-apps-in-v/)
+we created a complete command-line application using the built-in `flag`
+module. So we already have the directory structure like this:
+
+```text
+$ tree .
+.
+├── geometry
+│   ├── metadata.v
+│   ├── options.v
+│   ├── quadrilateral.v
+│   └── triangle.v
+├── geo.v
+└── v.mod
+```
+
+The `v.mod` file anchors the root of the project so that it becomes easier
+to use the `geometry` module. It also contains information about the project
+itself, like the version, description, license, and so on.
+
+The `geo.v` contains the main entry point of the application. It processes
```

Comment:
```suggestion
`geo.v` contains the main entry point of the application. It processes
```

Reply:
```suggestion
The file `geo.v` contains the main entry point of the application. It processes
```

Reply:
^ that is another alternative - starting the sentence with just `geo.v` seems weird